### PR TITLE
Database events

### PR DESCRIPTION
**Pull request by cmutel**

### CODE REVIEW

**Comment by cmutel** [comment]
Draft until tests for database copy and rename are added. These events are not new signals, but should be a combination of existing signals, and should just work™️, but need to be tested first.

You don't have to love this code, BTW 😛. But making it much better would require refactoring...  

Too a while because this is a totally different way of saving and interacting with the data, no SQLite here (yet).

File: bw2data/project.py [hunk 1]
```diff
@@ -571,23 +573,62 @@ def signal_dispatcher(
     """Not sure why this is necessary, but fails silently if call `add_revision` directly"""
     from bw2data import revisions
 
-    delta = revisions.generate_delta(old, new, operation)
+    try:
+        delta = revisions.generate_delta(old, new, operation)
+    except NoRevisionNeeded:
+        return
```
Comment (jsvgoncalves) [inline bw2data/project.py]: Why do we raise if we then just ignore it? Would it be better for the execution flow to just `return None` and act here accordingly?

Comment (cmutel) [reply bw2data/project.py]: Made this change in 3728040475ca3f10d9d092068b9a83d49fa1446e, but I don't love it - I find it cleaner to have the functions always have a single return signature. In any case it should be fixed now.

**Comment by cmutel** [comment]
Ready for review

**Comment by cmutel** [comment]
Rebased after #207 was merged.

**Comment by cmutel** [comment]
Moved back to draft after some tests are now failing.

File: bw2data/project.py (hunk 1 above)
Comment (jsvgoncalves) [reply bw2data/project.py]: I don't disagree with that, so I think the issue is something else upstream. In any case, this is not worth iterating on rn, not when we have 5K+ LOC changed.

**Review by jsvgoncalves — APPROVED** [review]
Overall LGTM, I will just merge this and we deal with issues later.

**Pull request merged**